openai_integration.py

```diff
@@ -0,0 +1,104 @@
+from dotenv import load_dotenv
+from datetime import datetime, timedelta
+import asyncio
+import openai
+import os
+
+# Load the environment variables from the .env file
+load_dotenv()
+
+# Counter for the number of requests made
+request_counter = 0
+
+# Last request time
+last_request_time = None

```

Comment:
you need to consider implementing a class for your project.
it is bad practice to have a variable outside of functions that you reach and change whenever you want.